indra/llcommon/llrand.cpp

```diff
@@ -58,9 +58,26 @@
  * to restore uniform distribution.
  */
 
-// gRandomGenerator is a stateful static object, which is therefore not
+// pRandomGenerator is a stateful static object, which is therefore not
 // inherently thread-safe.
-static thread_local LLRandLagFib2281 gRandomGenerator(LLUUID::getRandomSeed());
+//We use a pointer to not construct a huge object in the TLS space, sadly this is necessary
+// due to libcef.so on Linux being compiled with TLS model initial-exec (resulting in
+// FLAG STATIC_TLS, see readelf libcef.so). CEFs own TLS objects + LLRandLagFib2281 then will exhaust the
+// available TLS space, causing media failure.
```

Comment:
How does exhausted TLS space manifest? Would there be anything in the viewer log, or would we have to look in system logs?

We're investigating a Mac crash bug that may (or may not) be related to TLS consumption.